k4FWCore/include/k4FWCore/PodioDataSvc.h

```diff
@@ -40,36 +42,32 @@ class PodioDataSvc : public DataSvc {
   virtual StatusCode registerObject(std::string_view parentPath, std::string_view fullPath,
                                     DataObject* pObject) override final;
 
-  StatusCode readCollection(const std::string& collectionName, int collectionID);
+  StatusCode readCollection(const std::string& collectionName);
 
-  virtual const CollRegistry&       getCollections() const { return m_collections; }
-  podio::EventStore&                getProvider() { return m_provider; }
-  virtual podio::CollectionIDTable* getCollectionIDs() { return m_collectionIDs; }
+  const podio::Frame& getEventFrame() const { return m_eventframe; }
+  podio::Frame& getMetaDataFrame() { return m_metadataframe; }
```

Comment:
```suggestion
  const podio::Frame& getMetaDataFrame() { return m_metadataframe; }
```
Seems to be only used for writing (unless I missed something)

Reply:
Can we do some `friend` definition again to limit the exposure of mutable state here? Otherwise, this will be a source of (attempted) abuse and confusion.